src/CQRS/LeanCode.CQRS.AspNetCore/ServiceCollectionCQRSExtensions.cs

```diff
@@ -19,31 +19,39 @@ TypesCatalog handlersCatalog
     {
         serviceCollection.AddSingleton<ISerializer>(_ => new Utf8JsonSerializer(Utf8JsonSerializer.DefaultOptions));
 
-        var objectsSource = new CQRSObjectsRegistrationSource(contractsCatalog, handlersCatalog);
+        var objectsSource = new CQRSObjectsRegistrationSource(serviceCollection);
+        objectsSource.AddCQRSObjects(contractsCatalog, handlersCatalog);
 
         serviceCollection.AddSingleton(objectsSource);
-        serviceCollection.AddCQRSHandlers(objectsSource.Objects);
 
         serviceCollection.AddSingleton<RoleRegistry>();
         serviceCollection.AddScoped<IHasPermissions, DefaultPermissionAuthorizer>();
         serviceCollection.AddScoped<ICommandValidatorResolver, CommandValidatorResolver>();
 
-        return new CQRSServicesBuilder(serviceCollection);
+        return new CQRSServicesBuilder(serviceCollection, objectsSource);
     }
 }
 
 public class CQRSServicesBuilder
 {
-    private readonly IServiceCollection services;
+    public IServiceCollection services { get; }
```

Comment:
```suggestion
    public IServiceCollection Services { get; }
```

Convention, though I'm surprised your IDE didn't scream at you for leaving this all lowercase.